docs/source/developers/java/development.rst

```diff
@@ -137,3 +137,14 @@ This applies the style to all pom.xml files under the current directory or from
 .. _conbench: https://github.com/conbench/conbench
 .. _checkstyle: https://github.com/apache/arrow/blob/main/java/dev/checkstyle/checkstyle.xml
 .. _Apache Maven pom.xml guidelines: https://maven.apache.org/developers/conventions/code.html#pom-code-convention
+
+
+Error Prone
+===========
+
+In relation to code formatting and standards, ``Error Prone`` compiler analysis may use cache to avoid
+showing the warnings if attempted re-run of ``mvn clean install``. To avoid the cache to include the
+warnings, use ``mvn clean install -Ddevelocity.cache.local.enabled=false -Ddevelocity.cache.remote.enabled=false``
+instead.
+
+Also note that Error Prone must be disabled for auto generated content.
```

Comment:
```suggestion
Build Caching
=============

Build caching is done through Develocity (formerly Maven Enterprise).  To force
a build without the cache, run::

    mvn clean install -Ddevelocity.cache.local.enabled=false -Ddevelocity.cache.remote.enabled=false

This can be useful to make sure you see all warnings from ErrorProne, for example.

ErrorProne
==========

ErrorProne should be disabled for generated code.
```